apps/server/src/modules/oauth/service/oauth.service.spec.ts

```diff
@@ -0,0 +1,342 @@
+	const defaultUserId = '123456789';
+	const defaultScool: School = {
+		name: '',
+		_id: new ObjectId(),
+		id: '',
+		systems: new Collection<System>([]),
+	};
+	const defaultDecodedJWT = {
+		sub: '',
+		uuid: '1111',
+	};
+
+	const defaultUser: User = {
+		email: '',
+		roles: new Collection<Role>([]),
+		school: defaultScool,
+		_id: new ObjectId(),
+		id: defaultUserId,
+		createdAt: new Date(),
+		updatedAt: new Date(),
+		ldapId: '1111',
+		firstName: 'Test',
+		lastName: 'Testmann',
+	};
+	const defaultIservUser: User = {
+		email: '',
+		roles: new Collection<Role>([]),
+		school: defaultScool,
+		_id: new ObjectId(),
+		id: defaultUserId,
+		createdAt: new Date(),
+		updatedAt: new Date(),
+		ldapId: '3333',
+		firstName: 'iservTest',
+		lastName: 'Test',
+	};
+
+	const defaultSystemId = '987654321';
+	const defaultIservSystemId = '2222';
+	const defaultJWT =
+		'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyLCJ1dWlkIjoiMTIzIn0.H_iI0kYNrlAUtHfP2Db0EmDs4cH2SV9W-p7EU4K24bI';
+	const wrongJWT =
+		'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c';
+	const defaultTokenResponse: OauthTokenResponse = {
+		access_token: 'zzzz',
+		refresh_token: 'zzzz',
+		id_token: defaultJWT,
+	};
+	const defaultAxiosResponse: AxiosResponse<OauthTokenResponse> = {
+		data: defaultTokenResponse,
+		status: 0,
+		statusText: '',
+		headers: {},
+		config: {},
+	};
+
+	const defaultDecryptedSecret = 'IchBinNichtMehrGeheim';
+	const decodedJWTMock: IJwt = {
+		uuid: '1111',
+		sub: ' ',
+	};
+
+	const defaultResponse: OAuthResponse = {
+		idToken: defaultJWT,
+		logoutEndpoint: 'logoutEndpointMock',
+		provider: 'providerMock',
+		redirect: '',
+	};
+
+	const iservRedirectMock = `logoutEndpointMock?id_token_hint=${defaultJWT}&post_logout_redirect_uri=${
+		Configuration.get('HOST') as string
+	}/dashboard`;
+	const defaultIservSystem: System = {
+		type: 'iserv',
+		url: 'http://mock.de',
+		alias: `system`,
+		oauthConfig: {
+			clientId: '12345',
+			clientSecret: 'mocksecret',
+			tokenEndpoint: 'http://mock.de/mock/auth/public/mockToken',
+			grantType: 'authorization_code',
+			tokenRedirectUri: 'http://mockhost:3030/api/v3/oauth/testsystemId/token',
+			scope: 'openid uuid',
+			responseType: 'code',
+			authEndpoint: 'mock_authEndpoint',
+			provider: 'iserv',
+			logoutEndpoint: 'logoutEndpointMock',
+			issuer: 'mock_issuer',
+			jwksEndpoint: 'mock_jwksEndpoint',
+		},
+		_id: new ObjectId(),
+		id: '',
+		createdAt: new Date(),
+		updatedAt: new Date(),
+	};
+
+	beforeEach(async () => {
+		const module: TestingModule = await Test.createTestingModule({
+			imports: [HttpModule],
+			providers: [
+				OAuthService,
+				{
+					provide: Logger,
+					useValue: createMock<Logger>(),
+				},
+				{
+					provide: 'OAuthEncryptionService',
+					useValue: {
+						decrypt: () => {
```

Comment:
If possible, I would prefer to use createMock here, and set up the specific expected response for each specific testCondition.

You can avoid code repitition in the tests by defining a common setup method for shared preconditions that handle setting up the mocks as necessary.